File: 240923/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.py
from math import sqrt

a, b = map(int, input().split())

def digitNumber(n):
    digitsum = 0
    while n > 0:
        r = n % 10
        digitsum += r
        n = n // 10
    return digitsum

def primeNumber(n):
    if n < 2:
        return False
    for i in range(2, int(sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

result = 0
for j in range(a, b+1):
    if digitNumber(j) % 2 == 0 and primeNumber(j):
        result += 1

print(result)

File: 240923/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.c
#include <stdio.h>
#include <math.h>

int primeNumber(int n) {
    if (n <= 1) return 0;
    int value, i;
    value = (int)sqrt((double)n);
    for (i=2; i<=value; i++) {
        if (n % i == 0) return 0;
    }
    return 1;
}

int digitNum(int n) {
    int digitsum = 0, r;
    while(n > 0) {
        r = n % 10;
        digitsum += r;
        n /= 10;
    }
    return digitsum;
}

int main() {
    int a, b;
    scanf("%d %d", &a, &b);
    int result = 0, j;
    for (j=a; j<=b; j++) {
        if (primeNumber(j) == 1 && digitNum(j) % 2 == 0) {
            result += 1;
        }
    }
    printf("%d", result);
    return 0;
}